File: backend/core/src/modules/research-items/research-items.controller.ts
import { Body, Controller, Post, Req, UseGuards } from '@nestjs/common';
import { ProjectScopedCrudController } from '../../common/access/project-scoped-crud.controller';
import { ResearchItemsService } from './research-items.service';
import { ResearchItem } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('research-items')
export class ResearchItemsController extends ProjectScopedCrudController<ResearchItem> {
  constructor(service: ResearchItemsService) {
    super(service);
  }

  @Post('generate')
  generate(@Req() req: any, @Body() dto: any) {
    return (this.service as ResearchItemsService).generateForUser(req.user, dto);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/ResearchItemsController.java
package store.pjcloud.cowrite.core.controller;

import java.util.Map;
import java.util.UUID;
import org.springframework.web.bind.annotation.*;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudController;
import store.pjcloud.cowrite.core.entity.ResearchItem;
import store.pjcloud.cowrite.core.security.SecurityUtils;
import store.pjcloud.cowrite.core.service.AiService;
import store.pjcloud.cowrite.core.service.ResearchItemsService;

@RestController
@RequestMapping("/research-items")
public class ResearchItemsController extends ProjectScopedCrudController<ResearchItem> {
    private final ResearchItemsService researchItemsService;

    public ResearchItemsController(ResearchItemsService service) {
        super(service);
        this.researchItemsService = service;
    }

    @PostMapping("/generate")
    public ResearchItem generate(@RequestBody Map<String, Object> body) {
        UUID userId = SecurityUtils.requireUserId();
        String role = SecurityUtils.getRole();
        return researchItemsService.generateForUser(new AiService.UserContext(userId, role), body);
    }
}
